### PR TITLE
Fix WebSocket frame concatenation for Netty

### PR DESCRIPTION
**Pull request by kciesielski**
When handling incoming WebSocket frames, there was a missing case for a "non-final data frame and no data accumulated".
1. I added the case and moved frame concatenation to `WebSocketFrameConverters`, so it's shared between netty-sync and netty-cats
2. I also added cases where a binary frame comes after a text frame, this is specific to Netty (a `WebSocketContinuationFrame` is a binary frame that follows a text frame). 
3. I added test cases mentioned in 1. to other backends where they were missing: Vert.X, zio-http, http4s.

BTW it turns out http4s does frame concatenation by itself, regardless of our configuration.

After adding missing cases, there are issues with concatenation in zio-http and vert.x. The stream freezes after receiving first frame in the accumulation. This needs separate investigation.

### CODE REVIEW

File: server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala [hunk 1]
```diff
@@ -101,6 +101,8 @@ private[http4s] object Http4sWebSockets {
         case (None, f: WebSocketFrame.Pong)                                  => (None, Some(f))
         case (None, f: WebSocketFrame.Close)                                 => (None, Some(f))
         case (None, f: WebSocketFrame.Data[_]) if f.finalFragment            => (None, Some(f))
+        case (None, f: WebSocketFrame.Text)                                  => (Some(Right(f.payload)), None)
```
Comment (adamw) [inline server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: indeed this seems rather basic :D

Comment (adamw) [reply server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: but if http4s concatenates frames by itself, shouldn't we remove this? it's dead code anyway?

Comment (kciesielski) [reply server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: Ok, turns out that:
1. Http4s can be configured to not concatenate fragmented frames and not ignore pings/pongs on the level of `WebSocketBuilder2`:
```scala
      .withHttpWebSocketApp(wsb => Router("/" -> wsRoutes(wsb.withDefragment(false).withFilterPingPongs(false)).orNotFound)
```
2. This has no effect if used with Blaze, which always automatically defragments and filters pings pongs
3. It works with Ember though.

I'll remove these stages from `Http4sWebSockets` and add some documentation.

Comment (kciesielski) [reply server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: Or we can keep our own defragmentation and force `wsb.withDefragment(false)` to make this Tapir-native switch work on Ember by forcing passing through our concatenation :thinking:

Comment (adamw) [reply server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: Ah ... if it's optional, let's keep the code, and let the user of http4s decide. So I guess the code is good as-is

Comment (kciesielski) [reply server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala:104]: It has several bugs actually, as I discovered when I switched from Blaze to Ember in tests. Fixes are on the way in a separate PR then ;)

**Pull request merged**